lib/Jsrt/Jsrt.cpp

```diff
@@ -258,42 +257,26 @@ JsErrorCode CreateContextCore(_In_ JsRuntimeHandle runtimeHandle, _In_ bool crea
 #if ENABLE_TTD
         if (createUnderTimeTravel)
         {
-            if(threadContext->IsTTRequested && !threadContext->IsTTDInitialized())
-            {
-                if(threadContext->TTDInitializeTTDUriFunction == nullptr)
-                {
-                    AssertMsg(false, "Must set this callback function first!!!");
-                    return JsErrorCategoryUsage;
-                }
-
-                wchar_t* ttdlogStr = nullptr;
-                threadContext->TTDInitializeTTDUriFunction(threadContext->TTDUri, &ttdlogStr);
-                if(ttdlogStr == nullptr)
-                {
-                    AssertMsg(false, "OOM on path allocation!!!");
-                    return JsErrorFatal;
-                }
-
-#if ENABLE_TTD_DEBUGGING
-                threadContext->SetThreadContextFlag(ThreadContextFlagNoJIT);
-#endif
-
-                threadContext->InitTimeTravel(ttdlogStr, threadContext->IsTTRecordRequested, threadContext->IsTTDebugRequested, threadContext->TTSnapInterval, threadContext->TTSnapHistoryLength);
-
-                CoTaskMemFree(ttdlogStr);
-            }
-
             HostScriptContextCallbackFunctor callbackFunctor(context, &JsrtContext::OnScriptLoad_TTDCallback);
             threadContext->BeginCtxTimeTravel(context->GetScriptContext(), callbackFunctor);
 
-            if (threadContext->IsTTRecordRequested | threadContext->IsTTDebugRequested)
+#if TTD_DYNAMIC_DECOMPILATION_WORK_AROUNDS
+            if(threadContext->IsTTRecordRequested | threadContext->IsTTDebugRequested)
             {
                 //
                 //TODO: We currently force this into debug mode in record as well to make sure parsing/bytecode generation is same as during replay.
                 //      Later we will want to be clever during inflate and not do this.
                 //
+#ifdef _WIN32
                 context->GetScriptContext()->InitializeDebugging();
+#else
+                //
+                //TODO: x-plat does not like some parts of initiallize debugging so just set the flag we need 
+                //
+                context->GetScriptContext()->GetDebugContext()->SetDebuggerMode(Js::DebuggerMode::Debugging);
+#endif
```

Comment:
I had fixed SetDispatchProfile call in RegisterDebugThunk sometime back, are you still seeing the issue? Instead of this we should fix the issue, I think it should just be some initialization issue